toolkit/offchain/src/reserve/mod.rs

```diff
@@ -17,14 +19,35 @@ pub mod init;
 pub mod update_settings;
 
 #[derive(Clone, Debug)]
-pub(crate) struct ReserveData {
-	pub(crate) scripts: scripts_data::ReserveScripts,
-	pub(crate) auth_policy_version_utxo: OgmiosUtxo,
-	pub(crate) validator_version_utxo: OgmiosUtxo,
-	pub(crate) illiquid_circulation_supply_validator_version_utxo: OgmiosUtxo,
+pub struct ReserveData {
+	pub scripts: scripts_data::ReserveScripts,
+	pub auth_policy_version_utxo: OgmiosUtxo,
+	pub validator_version_utxo: OgmiosUtxo,
+	pub illiquid_circulation_supply_validator_version_utxo: OgmiosUtxo,
 }
 
-pub(crate) async fn get_reserve_data<
+impl ReserveData {
+	pub async fn get_reserve_settings<
+		T: QueryLedgerState + Transactions + QueryNetwork + QueryUtxoByUtxoId,
+	>(
+		&self,
+		ctx: &TransactionContext,
+		client: &T,
+	) -> Result<Option<(OgmiosUtxo, ReserveDatum)>, anyhow::Error> {
+		let validator_address = self.scripts.validator.address(ctx.network).to_bech32(None)?;
+		let validator_utxos = client.query_utxos(&[validator_address]).await?;
+
+		Ok(validator_utxos.into_iter().find_map(|utxo| {
```

Comment:
Is adversary restricted in anyway from transferring an UTXO with some spoofed datum that will cause this code retrieve his UTXO?

I think we should make sure, that UTXO contains Reserve Auth Policy token.